src/nbl/video/IOpenGL_LogicalDevice.h

```diff
@@ -230,6 +247,13 @@ class IOpenGL_LogicalDeviceBase
 
             core::smart_refctd_ptr<IDriverMemoryAllocation> buf;
         };
+        struct SRequestAllocate
+        {
+            static inline constexpr E_REQUEST_TYPE type = ERT_ALLOCATE;
+            using retval_t = IDeviceMemoryAllocator::SMemoryOffset;
+            IDeviceMemoryAllocator::SAllocateInfo allocateInfo;
+            core::bitflag<IPhysicalDevice::E_MEMORY_PROPERTY_FLAGS> memoryTypeIndexFlags;
+        };
```

Comment:
I see what you did, but its best to "allocate" inside createBuffer/createImage on GL, because you want to ensure `glStorageBuffer` and `glStorageImage` run before the user has a chance to bind the generated handles to anything

Reply:
so we shouldn't have an allocate command